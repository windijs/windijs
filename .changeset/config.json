{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [["@windijs/core", "windijs"]],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
