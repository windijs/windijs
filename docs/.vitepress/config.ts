import { defineConfig } from "vitepress";
import { mdRenderFilename } from "./markdown";

export default defineConfig({
  lang: "en-US",
  title: "Windi JS",
  cleanUrls: "without-subfolders",
  description: "The Utility-first CSS-in-JS Framework Powered by ES6 Proxy API and TypeScript.",
  markdown: {
    config(md) {
      md.use(mdRenderFilename);
    },
  },
  head: [
    ['link', { rel: "shortcut icon", type: "image/x-icon", href: "/windijs/favicon.ico" }]
  ],
  base: "/windijs/",
  themeConfig: {
    logo: "/logo.png",
    nav: [
      { text: "Guide", link: "/guide/what-is-windijs" },
      { text: "API", link: "/api" },
      { text: "Repl", link: "/repl" },
      { text: "Examples", link: "https://github.com/windijs/windijs/tree/main/examples" },
    ],
    sidebar: [
      {
        text: "Guide",
        collapsible: false,
        items: [
          {
            text: "What is Windi JS?",
            link: "/guide/what-is-windijs",
          },
          {
            text: "Getting Started",
            link: "/guide/getting-started",
          },
          {
            text: "Configuration",
            link: "/guide/configuration",
          },
          {
            text: "Optimization",
            link: "/guide/optimization",
          },
        ],
      },
      {
        text: "Customization",
        collapsible: false,
        items: [
          {
            text: "Theme",
            link: "/customization/theme",
          },
          {
            text: "Utility",
            link: "/customization/utility",
          },
          {
            text: "Variant",
            link: "/customization/variant",
          },
          {
            text: "Handler",
            link: "/customization/handler",
          },
          {
            text: "Loader",
            link: "/customization/loader",
          },
          {
            text: "Namer",
            link: "/customization/namer",
          },
          {
            text: "Plugins",
            link: "/customization/plugins",
          },
        ],
      },
      {
        text: "Integrations",
        collapsible: false,
        items: [
          {
            text: "API",
            link: "/integrations/api",
          },
          // {
          //   text: "CLI",
          //   link: "/integrations/cli",
          // },
          {
            text: "Vite",
            link: "/integrations/vite",
          },
          // {
          //   text: "Rollup",
          //   link: "/integrations/rollup",
          // },
          // {
          //   text: "Webpack",
          //   link: "/integrations/webpack",
          // },
        ],
      },
      {
        text: "Posts",
        collapsible: false,
        items: [
          {
            text: "How Windi JS Works",
            link: "/posts/how-windijs-works",
          },
          {
            text: "Ways of Using Windi JS",
            link: "/posts/ways-of-using-windijs",
          },
        ],
      },
    ],
    editLink: {
      pattern: "https://github.com/windijs/windijs/edit/main/docs/:path",
      text: "Edit this page on GitHub",
    },
    socialLinks: [
      { icon: "discord", link: "https://discord.gg/nJux6Gb52m" },
      { icon: "github", link: "https://github.com/windijs/windijs" },
    ],
    footer: {
      message: "Released under the MIT License.",
      copyright: "Copyright Â© 2022-present Raven Satir",
    },
  },
});
