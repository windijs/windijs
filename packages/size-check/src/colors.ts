import { windiColors } from "@windijs/colors";

export { windiColors };
