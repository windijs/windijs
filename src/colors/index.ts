export * from "./base";
export * from "./web";
export * from "./windi";
export * from "./apple";
export * from "./bulma";
export * from "./material";
export * from "./tailwind";
export * from "./bootstrap";
