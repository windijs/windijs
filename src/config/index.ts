export * from "./colors";
export * from "./bg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./positioning";
export * from "./interactivity";
export * from "./typography";
