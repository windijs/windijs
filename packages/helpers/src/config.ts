import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
