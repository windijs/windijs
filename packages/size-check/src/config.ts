import { paddingConfig } from "@windijs/config";

export { paddingConfig };
