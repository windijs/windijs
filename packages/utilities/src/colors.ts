import { baseColors, windiColors } from "@windijs/colors";

export default { ...baseColors, ...windiColors };
