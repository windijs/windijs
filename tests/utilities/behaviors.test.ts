import { boxDecorationBreakConfig, imageRenderingConfig, listStylePositionConfig, listStyleTypeConfig, opacityConfig, overflowConfig, overscrollConfig } from "config";
import { buildImageRendering, buildPlaceholder, colorHandler, configHandler, createUtility, cssHandler } from "utilities";

import { prop } from "helpers";
import { windiColors } from "colors";

test("Box Decoration Break", () => {
  const decoration = createUtility("decoraton").use(configHandler(boxDecorationBreakConfig, [prop`-webkit-box-decoration-break`, prop`box-decoration-break`])).init();

  expect(decoration.slice.css).toMatchSnapshot();
  expect(decoration.clone.css).toMatchSnapshot();
});

test("Image Rendering", () => {
  const image = createUtility("image").case("render", configHandler(imageRenderingConfig, buildImageRendering)).init();
  expect(image.render.auto.css).toMatchSnapshot();
  expect(image.render.pixel.css).toMatchSnapshot();
  expect(image.render.edge.css).toMatchSnapshot();
});

test("List", () => {
  const list = createUtility("list")
    .use(configHandler(listStyleTypeConfig, "listStyleType"))
    .use(configHandler(listStylePositionConfig, "listStylePosition"))
    .init();
  expect(list.disc.css).toMatchSnapshot();
  expect(list.zeroDecimal.css).toMatchSnapshot();
  expect(list.upperRoman.css).toMatchSnapshot();
  expect(list.inside.css).toMatchSnapshot();
  expect(list.outside.css).toMatchSnapshot();
});

test("Overflow", () => {
  const overflow = createUtility("overflow")
    .use(configHandler(overflowConfig, "overflow"))
    .case("truncate", cssHandler({ overflow: "hidden", "-o-text-overflow": "ellipsis", textOverflow: "ellipsis", whiteSpace: "nowrap" }))
    .case("ellipsis", cssHandler({ "-o-text-overflow": "ellipsis", textOverflow: "ellipsis" }))
    .case("x", configHandler(overflowConfig, "overflowX"))
    .case("y", configHandler(overflowConfig, "overflowY"))
    .init();

  expect(overflow.scroll.css).toMatchSnapshot();
  expect(overflow.x.hidden.css).toMatchSnapshot();
  expect(overflow.y.visible.css).toMatchSnapshot();
  expect(overflow.truncate.css).toMatchSnapshot();
  expect(overflow.ellipsis.css).toMatchSnapshot();
  expect(overflow.clip.css).toMatchSnapshot();
});

test("OverScroll", () => {
  const overscroll = createUtility("overscroll")
    .use(configHandler(overscrollConfig, "overscrollBehavior"))
    .case("x", configHandler(overscrollConfig, "overscrollBehaviorX"))
    .case("y", configHandler(overscrollConfig, "overscrollBehaviorY"))
    .init();

  expect(overscroll.contain.css).toMatchSnapshot();
  expect(overscroll.x.none.css).toMatchSnapshot();
  expect(overscroll.y.auto.css).toMatchSnapshot();
});

test("Placeholder", () => {
  const placeholder = createUtility("placeholder")
    .use(colorHandler(windiColors, buildPlaceholder))
    .case("opacity", configHandler(opacityConfig, prop`--w-placeholder-opacity`))
    .init();

  expect(placeholder.amber[300].css).toMatchSnapshot();
  expect(placeholder.amber[300].opacity(20).css).toMatchSnapshot();
  expect(placeholder.opacity[20].css).toMatchSnapshot();
});
