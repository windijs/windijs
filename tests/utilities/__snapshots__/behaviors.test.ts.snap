// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Decoration Break 1`] = `
Object {
  "-webkit-box-decoration-break": "slice",
  "box-decoration-break": "slice",
}
`;

exports[`Box Decoration Break 2`] = `
Object {
  "-webkit-box-decoration-break": "clone",
  "box-decoration-break": "clone",
}
`;

exports[`Image Rendering 1`] = `
Object {
  "imageRendering": "auto",
}
`;

exports[`Image Rendering 2`] = `
Object {
  "-ms-interpolation-mode": "nearest-neighbor",
  "imageRendering": Array [
    "-webkit-optimize-contrast",
    "-moz-crisp-edges",
    "-o-pixelated",
    "pixelated",
  ],
}
`;

exports[`Image Rendering 3`] = `
Object {
  "imageRendering": "crisp-edges",
}
`;

exports[`List 1`] = `
Object {
  "listStyleType": "disc",
}
`;

exports[`List 2`] = `
Object {
  "listStyleType": "decimal-leading-zero",
}
`;

exports[`List 3`] = `
Object {
  "listStyleType": "upper-roman",
}
`;

exports[`List 4`] = `
Object {
  "listStylePosition": "inside",
}
`;

exports[`List 5`] = `
Object {
  "listStylePosition": "outside",
}
`;

exports[`OverScroll 1`] = `
Object {
  "overscrollBehavior": "contain",
}
`;

exports[`OverScroll 2`] = `
Object {
  "overscrollBehaviorX": "none",
}
`;

exports[`OverScroll 3`] = `
Object {
  "overscrollBehaviorY": "auto",
}
`;

exports[`Overflow 1`] = `
Object {
  "overflow": "scroll",
}
`;

exports[`Overflow 2`] = `
Object {
  "overflowX": "hidden",
}
`;

exports[`Overflow 3`] = `
Object {
  "overflowY": "visible",
}
`;

exports[`Overflow 4`] = `
Object {
  "-o-text-overflow": "ellipsis",
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`Overflow 5`] = `
Object {
  "-o-text-overflow": "ellipsis",
  "textOverflow": "ellipsis",
}
`;

exports[`Overflow 6`] = `
Object {
  "overflow": "clip",
}
`;

exports[`Placeholder 1`] = `
Object {
  "&:-ms-input-placeholder": Object {
    "--w-placeholder-opacity": "1",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-moz-placeholder": Object {
    "--w-placeholder-opacity": "1",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-ms-input-placeholder": Object {
    "--w-placeholder-opacity": "1",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-webkit-input-placeholder": Object {
    "--w-placeholder-opacity": "1",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::placeholder": Object {
    "--w-placeholder-opacity": "1",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
}
`;

exports[`Placeholder 2`] = `
Object {
  "&:-ms-input-placeholder": Object {
    "--w-placeholder-opacity": "0.2",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-moz-placeholder": Object {
    "--w-placeholder-opacity": "0.2",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-ms-input-placeholder": Object {
    "--w-placeholder-opacity": "0.2",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::-webkit-input-placeholder": Object {
    "--w-placeholder-opacity": "0.2",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
  "&::placeholder": Object {
    "--w-placeholder-opacity": "0.2",
    "color": "rgba(252, 211, 77, var(--w-placeholder-opacity))",
  },
}
`;

exports[`Placeholder 3`] = `
Object {
  "--w-placeholder-opacity": "0.2",
}
`;
